scipy/optimize/_milp.py

```diff
@@ -322,7 +322,7 @@ def milp(c, *, integrality=None, bounds=None, constraints=None, options=None):
 
     >>> A = np.array([[-1, 1], [3, 2], [2, 3]])
     >>> b_u = np.array([1, 12, 12])
-    >>> b_l = np.full_like(b_u, -np.inf)
+    >>> b_l = np.full_like(b_u, -np.inf, dtype=float)
```

Comment:
@mdhaber git blame points at you, so I'd ask about this change. Without it, running the example from the `milp` docstring generates a warning from numpy; adding a dtype above changes the result from `[1.0, 2.0]` to `[2.0, 2.0]`:

```
In [1]: import numpy as np

In [2]: c = -np.array([0, 1])

In [3]: A = np.array([[-1, 1], [3, 2], [2, 3]])
   ...: >>> b_u = np.array([1, 12, 12])
   ...: >>> b_l = np.full_like(b_u, -np.inf)
/home/br/mambaforge/envs/scipy-dev/lib/python3.10/site-packages/numpy/core/numeric.py:407: RuntimeWarning: invalid value encountered in cast
  multiarray.copyto(res, fill_value, casting='unsafe')

In [4]: from scipy.optimize import LinearConstraint
   ...: >>> constraints = LinearConstraint(A, b_l, b_u)

In [5]: integrality = np.ones_like(c)

In [6]: from scipy.optimize import milp
   ...: >>> res = milp(c=c, constraints=constraints, integrality=integrality)
   ...: >>> res.x
Out[6]: array([1., 2.])
```